packages/rest/src/rest.server.ts

```diff
@@ -338,20 +343,42 @@ export class RestServer extends Context implements Server, HttpServerLike {
   }
 
   protected _setupHandlerIfNeeded() {
-    // TODO(bajtos) support hot-reloading of controllers
-    // after the app started. The idea is to rebuild the HttpHandler
-    // instance whenever a controller was added/deleted.
-    // See https://github.com/strongloop/loopback-next/issues/433
     if (this._httpHandler) return;
+    if (this._controllersAndRoutesView == null) {
+      // Watch for binding events
+      this._controllersAndRoutesView = this.createView(
+        binding =>
+          filterByKey(RestBindings.API_SPEC.key)(binding) ||
+          (filterByKey(/^(controllers|routes)\..+/)(binding) &&
+            // Exclude controller routes to avoid circular events
```

Comment:
Could you explain more about the `CONTROLLER_ROUTE`? I also feel confused for the test related to it. 
For example, when do we tag a binding with `REST_ROUTE` and when tag `CONTROLLER_ROUTE`? Do they overlap? And why `CONTROLLER_ROUTE` need to be excluded here?

Reply:
We discover controller bindings and add routes for each controller as a binding tagged with `CONTROLLER_ROUTE`. The `routes.*` also contain routes from non-controllers. This tag allows us to rebuild controller routes when a controller is added/removed.

Reply:
I see thanks for the explanation!